### DISCUSSION TITLE
Feature Request…

### DISCUSSION THREAD

**Discussion by Addonnis** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#366] [Ideas]
I think a timer since last action would be cool. 

Ideally it’d be cool to see a countdown for the bubbles, because I want to know how long it will be before the bubbles stop. 

I use 30 min bubble window to time my sessions in the spa and having a countdown would be awesome. 

I recognise I don’t have dev skill to contribute this so just floating the idea if anyone was interested too!

**Comment by visualapproach** [comment]
Great idea! Do you think it should count down last ten seconds, or once every minute or what´s your thought?

**Reply by Addonnis** [reply]
I was just thinking a simple clock/timer in the web app that starts counting down minutes & seconds when the bubbles are enabled! 

Even if it was a generic “time since last interaction”, it would give an indication on when the pump will turn off (I believe it is 3 days)

Having the “count down” show in full on the pump display for bubbles would be awesome (30:00, 29:00, 28:00 etc) but I expect this would be more complex to develop.

A “last 1 minute before bubbles turn off” beep from the unit would even be appreciated too

**Reply by visualapproach** [reply]
Update: no work done on this yet, but not forgotten :-)

**Reply by visualapproach** [reply]
@Addonnis this feature is implemented in todays dev version. Not exactly as described but close. You need to change `notify = false;` to `notify = true;` in `config.h`.
Default: 32 secs prior to a scheduled event (bubbles, pump, heater, jets) it will beep and show `-32-`in the display. Then it will do this at 16, 8, 4, 2 and 1 seconds. The pitch will go up for every beep.

**Reply by Addonnis** [reply]
Apologies for the delay coming back to you here - My spa has been down for a little while, heater died and I ended up buying a new pump. Very grateful for this feature, thank you very much. This is better than what I was expecting!

**Comment by Bishbosh02** [comment]
For the timer I would do this all-in Home Assistant / Node Red.......

**Comment by AttiMM** [comment]
Hi @visualapproach ,
thank you so much for your great efforts for making easier and more user friendly using our tub! :-)
I can confirm that your module is working fine with our new Maldives with pump S200102!

Reading about virtual temperature feature I was thinking about a new feature, where the user can set a target date & time and the target temperature. With this feature the user can set that he/she wants to have a tub full of 39 °C water at 6 pm tomorrow for example.
If that would be possible connecting an external temperature sensor to your control unit (is it possible?) the control unit would have exact information about the external temperature (and its changes in the past) and the current temp of water in tub (it could run the pump in every 1 hour for 2 minutes to read the exact temperature of water). 
Based on this data the control unit could calculate the starting time of the heating for reaching the desired temperature of water at the set time.

What do you think about this idea?
Attila

**Reply by visualapproach** [reply]
My pleasure @AttiMM it is satisfying to be able to improve people's lives a tiny bit.

About your request, yeah I was thinking about the same thing a long time ago and forgot about it. I think it's possible and with today's energy prices it may even save some €$£. I haven't been able to connect an extra thermometer due to limitations on the remaining pins. I can think of a few more involved ways like converting signal to analog but it would be easier to just build a new device that sends the ambient temp over mqtt or websockets. Or using a met office services on the net to push the data to the device. Ideally you would want to know the real power from the heaters, since it will vary with the mains voltage. But I'll estimate the resulting time difference considering that and see if is acceptable.

Thank you

**Reply by AttiMM** [reply]
"met office services" - what a great idea! There's no need any other device and we can even get a forecast which can be used in calculation of needed heating time!
For example this one: https://open-meteo.com/en/docs#api_form
